### PR TITLE
Add functionality to diverse lore feat

### LINKED ISSUE #7419: FR: Diverse Lore to add -2 as an option to Roll Dialogue
It would be nice if the diverse lore feat added a -2 check box for skill checks for thaumaturges that have diverse lore as a feat. 

the feat should grant this option in the roll dialogue. 

https://2e.aonprd.com/Feats.aspx?ID=3699

### PR DESCRIPTION
**Pull request by bcronin90**
Adds a toggle to enable to -2 penalty on an Esoteric Lore check with Diverse Lore

Closes #7419

### CODE REVIEW

File: packs/data/feats.db/diverse-lore.json [hunk 1]
```diff
@@ -19,7 +19,24 @@
...
+                "option": "diverse-lore-active",
+                "toggleable": true
+            },
+            {
+                "key": "FlatModifier",
+                "label": "Diverse Lore",
+                "predicate": [
+                    "diverse-lore-active"
+                ],
+                "selector": "esoteric-lore",
```
Comment (simonward) [inline packs/data/feats.db/diverse-lore.json:36]:
```suggestion
                "selector": [
                    "esoteric",
                    "esoteric-lore",
                    "lore-esoteric"
                ],
```

File: packs/data/feats.db/diverse-lore.json [hunk 2]
```diff
@@ -19,7 +19,24 @@
...
+            {
+                "domain": "skill-ckeck",
+                "key": "RollOption",
+                "label": "Diverse Lore",
+                "option": "diverse-lore-active",
+                "toggleable": true
+            },
+            {
+                "key": "FlatModifier",
+                "label": "Diverse Lore",
```
Comment (simonward) [inline packs/data/feats.db/diverse-lore.json:32]:
```suggestion
```

File: packs/data/feats.db/diverse-lore.json [hunk 3]
```diff
@@ -19,7 +19,24 @@
         "prerequisites": {
             "value": []
         },
-        "rules": [],
+        "rules": [
+            {
+                "domain": "skill-ckeck",
+                "key": "RollOption",
+                "label": "Diverse Lore",
```
Comment (simonward) [inline packs/data/feats.db/diverse-lore.json:26]:
```suggestion
```

File: packs/data/feats.db/diverse-lore.json (hunk 2 above)
Comment (bcronin90) [reply packs/data/feats.db/diverse-lore.json:32]: Just for my understanding, aren't these labels necessary to have a readable text in the roll options dialog?

File: packs/data/feats.db/diverse-lore.json [hunk 4]
```diff
@@ -19,7 +19,24 @@
         "prerequisites": {
             "value": []
         },
-        "rules": [],
+        "rules": [
+            {
+                "domain": "skill-ckeck",
```
Comment (stwlam) [inline packs/data/feats.db/diverse-lore.json:24]:
```suggestion
                "domain": "lore-skill-ckeck",
```

File: packs/data/feats.db/diverse-lore.json [hunk 5]
```diff
@@ -19,7 +19,24 @@
...
+                "key": "RollOption",
+                "label": "Diverse Lore",
+                "option": "diverse-lore-active",
```
Comment (stwlam) [inline packs/data/feats.db/diverse-lore.json:27]:
```suggestion
                "option": "diverse-lore",
```

File: packs/data/feats.db/diverse-lore.json [hunk 6]
```diff
@@ -19,7 +19,24 @@
...
+                "toggleable": true
+            },
+            {
+                "key": "FlatModifier",
+                "label": "Diverse Lore",
+                "predicate": [
+                    "diverse-lore-active"
```
Comment (stwlam) [inline packs/data/feats.db/diverse-lore.json:34]:
```suggestion
                    "diverse-lore"
```

**Comment by stwlam** [comment]
I'd prefer to keep this as a default-disabled penalty rather than a toggle.

File: packs/data/feats.db/diverse-lore.json (hunk 2 above)
Comment (simonward) [reply packs/data/feats.db/diverse-lore.json:32]: If a label is omitted, the name of the rule element's parent item is used instead.

**Comment by bcronin90** [comment]
> I'd prefer to keep this as a default-disabled penalty rather than a toggle.

More questions, if you don't mind: How would "default-disabled" work? Is there like an `enabled` option for FlatModifier?

File: packs/data/feats.db/diverse-lore.json (hunk 6 above)
Comment (bcronin90) [reply packs/data/feats.db/diverse-lore.json:32]: Another question: Would this not lead to the penalty always being applied as it checks for `diverse-lore`, which is the slug of the feat?


Sorry if this is much ado about such a tiny change, but I'm trying to learn the general principles at work here.

**Comment by stwlam** [comment]
> More questions, if you don't mind: How would "default-disabled" work? Is there like an `enabled` option for FlatModifier?

Simply adding a predicate that won't be satisfied by a toggle (or any other way) will keep the penalty disabled until manually overridden from the roll dialog.

File: packs/data/feats.db/diverse-lore.json (hunk 6 above)
Comment (stwlam) [reply packs/data/feats.db/diverse-lore.json:32]: A roll option for a feat always starts with "feat:"

**Comment by bcronin90** [comment]
Thanks for all the hints. This should be it then

File: packs/data/feats.db/diverse-lore.json [hunk 7]
```diff
@@ -19,7 +19,20 @@
...
         },
-        "rules": [],
+        "rules": [
+            {
+                "key": "FlatModifier",
+                "predicate": [
+                    "diverse-lore"
+                ],
+                "selector": [
+                    "esoteric-lore",
```
Comment (simonward) [inline packs/data/feats.db/diverse-lore.json:29]: Sorting these is generally preferred.

File: packs/data/feats.db/diverse-lore.json [hunk 8]
```diff
@@ -19,7 +19,20 @@
...
+                    "lore-esoteric",
+                    "esoteric"
```
Comment (TikaelSol) [inline packs/data/feats.db/diverse-lore.json:31]:
```suggestion
                    "esoteric",
                    "esoteric-lore",
                    "lore-esoteric"
```

**Pull request merged**